lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1721,6 +1722,15 @@ def handle_joining_data_node(self) -> None:
         else:
             self._start_opensearch_event.emit(ignore_lock=True)
 
+    def put_or_update_security_index_initialized(self, event: EventBase) -> None:
```

Comment:
nit - in general `put` means "create or update"
```suggestion
    def put_security_index_initialized(self, event: EventBase) -> None:
```